addons/mass_mailing_sms/views/mailing_mailing_views.xml

```diff
@@ -98,12 +98,18 @@
                 <attribute name="attrs">{'invisible': ['|',('mailing_type', '!=', 'mail'),('state', 'in', ('draft','test'))]}</attribute>
             </xpath>
             <!-- Form -->
+            <xpath expr="//field[@name='subject']" position="attributes">
+                <attribute name="attrs">{'invisible': [('mailing_type', '!=', 'mail')], 'required': [('mailing_type', '=', 'mail')]}</attribute>
+            </xpath>
+            <xpath expr="//field[@name='mailing_type']" position="after">
```

Comment:
I would put it after the original "subject" instead to make sure they're placed at the same spot in the form.
(Since they're basically the same field).